modules/core/src/v2/recovery/initiate.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * @prettier
+ */
+
+import * as utxolib from '@bitgo/utxo-lib';
+import { BitGo } from '../../bitgo';
+import * as stellar from 'stellar-sdk';
+
+interface ValidateKeyOptions {
+  key: string;
+  source: string;
+  passphrase?: string;
+  isUnsignedSweep: boolean;
+  isKrsRecovery: boolean;
+}
+
+export interface InitiateRecoveryOptions {
+  userKey: string;
+  backupKey: string;
+  bitgoKey?: string; // optional for xrp recoveries
```

Comment:
I considered that as well but decided the PR was already large enough. We can do it in a follow-up.

Reply:
Ideally all of this code should move to the WRW anyway, which is the only place that needs it.

Reply:
> Ideally all of this code should move to the WRW anyway, which is the only place that needs it.

I was thinking a new `@bitgo/recovery` package might be a good place for this actually, because I think some similar code may be needed in https://github.com/BitGo/key-recovery-service-v2 or bgadmin.